pkgs/by-name/sq/sqruff/package.nix

```
@@ -0,0 +1,48 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  installShellFiles,
+  stdenv,
+  darwin,
+  rust-jemalloc-sys,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "sqruff";
+  version = "v0.17.0";
+
+  src = fetchFromGitHub {
+    owner = "quarylabs";
+    repo = "sqruff";
+    rev = "refs/tags/${version}";
+    hash = "sha256-uUtbVf4U59jne5uORXpyzpqhFQoviKi2O9KQ5s1CfhU=";
+  };
+
+  cargoHash = "sha256-LnP03JEyTXVACJ5LiTPeCYEFy3h4YUgQiaNxm9alpZA=";
+
+  # Requires nightly features (feature(let_chains) and feature(trait_upcasting))
+  RUSTC_BOOTSTRAP = true;
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  buildInputs = [
+    rust-jemalloc-sys
+  ] ++ lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.CoreServices ];
+
+  # Patch the tests to find the binary
+  postPatch = ''
+    substituteInPlace crates/cli/tests/ui.rs \
+      --replace-fail \
+      'config.program.program = format!("../../target/{profile}/sqruff").into();' \
+      'config.program.program = "../../target/${stdenv.hostPlatform.rust.cargoShortTarget}/release/sqruff".into();'
+  '';
+
+  meta = {
```

```suggestion
  '';
  
  passtrhu = {
    updateScript = nix-update-script { };
    tests.versions = testers.testVersion { package = sqruff; };
  };

  meta = {
```
This will make our lives easier.